tsl/src/compression/api.c

```diff
@@ -1075,128 +1059,682 @@ tsl_get_compressed_chunk_index_for_recompression(PG_FUNCTION_ARGS)
 		PG_RETURN_NULL();
 }
 
+static bool
+is_segment_exists_in_chunk(Chunk *uncompressed_chunk, RowDecompressor *decompressor,
+						   CompressedSegmentInfo **current_segment, int nsegmentby_cols,
+						   HeapTuple *compressed_tuple)
+{
+	/* build scankeys for segmentby columns */
+	ScanKeyData *scankeys = palloc0(nsegmentby_cols * sizeof(ScanKeyData));
```

Comment:
Not strictly necessary, but maybe you want to release the memory since you know it will not be needed after the scan.

Reply:
Fixed.